### DISCUSSION TITLE
Library for creating UBL files

### DISCUSSION THREAD

**Discussion by Kurounin** [TecsiAron/ANAF-API-Client-PHP#5] [General]
Hi,

Sorry if this is off-topic, but I was wondering if you could share what library you're using for creating UBL files or if you rolled your own. I found these 2 that look promising
https://github.com/num-num/ubl-invoice
https://github.com/josemmo/einvoicing

although I'm not familiar enough to know if they would work. [einvoicing](https://github.com/josemmo/einvoicing) seems to support CIUS-RO.

Any insight you can provide would be appreciated.
Thanks

**Comment by TecsiAron** [comment]
Heya,

Using num-num/ubl-invoice. It was mostly painless but it did require some manual changes, https://github.com/num-num/ubl-invoice/pull/54 comes to mind but I cant say for sure if it was the only one.
I can provide you with a sample if you leave an e-mail address or Discord ID. It will require heavy modification since it's pretty tightly integrated with our systems, but it might be a good jumping off point.

**Reply by Kurounin** [reply]
Thanks for the quick reply. My discord username is kurounin4life
